pkg/component/component.go

```diff
@@ -0,0 +1,229 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package component
+
+import (
+	"fmt"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+)
+
+// Unit is a single unit that needs to be running inside a component.
+type Unit struct {
+	ID     string
+	Type   client.UnitType
+	Config map[string]interface{}
```

Comment:
Should we encapsulate the configuration in a type alias to ensure we do not leak that anywhere in the log?

```golang
type UnitConfig map[string]interface{}

func (UnitConfig) String() string {
	return "Unit Configuration: redacted"
}
```

Reply:
I assume at this point all substitutions for variables that hold sensitive information would be computed. We will have no way of knowing if there is sensitive information in the configuration.

At the moment this PR doesn't add any CLI to take a policy and specs and generate a component model. Probably should look at maybe doing that in a different PR. Would prefer not to expand on this one anymore.

Reply:
Good point, I just want to reduce risk of just sending that map[string] to debug.